Add GUI Support

crazygmr101 opened the pull request:
Merge GUI Support Into master branch.

crazygmr101 committed:
Code Cleanup

crazygmr101 committed:
Full GUI

Console has stayed for logging, for now. All input is done through the game now. Force jump logic is turned off temporarily. (Board.java:94).

crazygmr101 committed:
Improved Graphics

dannash987 commented:
![image](https://user-images.githubusercontent.com/22755628/49773308-8f1f9e80-fcbe-11e8-9393-91af56ced20a.png)

crazygmr101 committed:
Revert "Added support for more than 10 moves. Added difficulty selector."

This reverts commit fe7e4bb47abb06343dc3e0e4776241c5aaacb2a5.

crazygmr101 committed:
Added support for more than 10 moves. Added difficulty selector.

crazygmr101 committed:
Javadoc

Added javadoc comments for all files. Changed Board.checkersAboutToKing():434 so the method actually works and didn't just return 0

crazygmr101 committed:
Created JavaDocs

crazygmr101 committed:
Merge branch 'docs' into gui-dev
